find.go

```diff
@@ -255,6 +264,19 @@ func (s *server) doFind(ctx context.Context, method, source string, req *url.URL
 	defer cancel()
 
 	var count int32
+
+	// if this is a get method - extract cid / multihash and add it as a shard key.
+	// TODO: how do we handle post methods?
+	var shardKey string
+	if method == http.MethodGet {
+		// cid.Decode can parse both cid and mutihash. If it's a multihash it would be treated as a base58 string.
+		c, err := cid.Decode(extractShardingKey(req.Path))
+		if err != nil {
+			log.Errorw("Failed to parse cid / multihash from the request. Not setting shard key.", "err", err)
```

Comment:
If a shard key was included, but fails to decode, then it seems like this should be an error the we do not continue from.
```suggestion
			return nil, fmt.Errorf("failed to parse cid / multihash from the request: %w", err)
```